### PR TITLE
daemon: don't wait for presence of unused CNC CRD

### LINKED ISSUE #27683: Agent shouldn't wait for `CiliumNodeConfig` CRD
Currently, cilium-agent waits for the `CiliumNodeConfig` CRD to be present in the cluster because it's included as one of the agent CRDs, but the agent doesn't actually operate on this CRD, it's only used in the `config` init container: https://github.com/cilium/cilium/blob/dcef00d653e1f9685e5bdadb1304d9b8679e44e6/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L402-L408

and the `build-config` CLI command already tolerates the CRD not existing: https://github.com/cilium/cilium/blob/dcef00d653e1f9685e5bdadb1304d9b8679e44e6/pkg/option/resolver/resolver.go#L238-L240

Can we remove the wait so that the agent can still function if the CRD is not present in the cluster? From an operator's perspective, I think there are several good reasons to not install this CRD:
- I don't want users to be able to override cilium config easily
- It's an alpha API, and I don't want to risk upgrades
- I want to reduce the API surface for its own sake

I don't think this requires any new flags. Just need to remove this line: https://github.com/cilium/cilium/blob/dcef00d653e1f9685e5bdadb1304d9b8679e44e6/pkg/k8s/synced/crd.go#L46

### PR DESCRIPTION
**Pull request by akhilles**
Currently, cilium-agent waits for the presence of the CiliumNodeConfig CRD but it doesn't need to. The commit allows for the agent to function even if the CRD is not present in the cluster.

Fixes: #27683

```release-note
daemon: don't wait for presence of unused CiliumNodeConfig CRD
```

### CODE REVIEW

**Review by youngnick — APPROVED** [review]

**Comment by akhilles** [comment]
The multi-pool-ipam-conformance-test test failure seems to be a known flake: https://github.com/cilium/cilium/issues/27417. @youngnick could you please re-run the tests?

Also, I'm not sure why this change is blocked by "needs-release-note-label". I did add a release note to the PR description.

**Comment by julianwiedmann** [comment]
@joestringer this PR has been sitting for a while. Do you think someone besides @squeed could provide the needed review? Or should we proceed to get CI green and :ship: ?

**Comment by akhilles** [comment]
Triage of test failures:

[Conformance E2E (ci-e2e)](https://github.com/cilium/cilium/actions/runs/6198047511/job/16827764982):
```
Test [echo-ingress-mutual-auth-spiffe]:
  ❌ echo-ingress-mutual-auth-spiffe/pod-to-pod/curl-ipv4-3: cilium-test/client2-646b88fb9b-srtq2 (10.244.3.242) -> cilium-test/echo-same-node-775456cfcf-4rxjx (10.244.3.208:8080)
```
This looks like https://github.com/cilium/cilium/issues/26351.

[Conformance AKS (ci-aks)](https://github.com/cilium/cilium/actions/runs/6198047379/job/16827803448), not marked as required:
```
connectivity test failed: failed to fetch cilium runtime config: error dialing backend: read unix @->/tunnel-uds/proxysocket: read: connection reset by peer
Error: Process completed with exit code 1.
```

This looks like https://github.com/cilium/cilium/issues/27989.

**Comment by joestringer** [comment]
> @joestringer this PR has been sitting for a while. Do you think someone besides @squeed could provide the needed review? Or should we proceed to get CI green and 🚢 ?

The key question I don't know the answer to is this: If the daemon doesn't wait for CNC, does that introduce some race condition or other issue with applying that configuration? Or was there never a reason for the agent to wait for this resource in the first place?

**Comment by squeed** [comment]
A fair question. No, we do not need to wait for the CiliumNodeConfig. @youngnick, how did you find yourself in this situation?

The command `cilium build-config`, which is what actually consumes these resources, is tolerant of the CRD not existing.

**Review by squeed — APPROVED** [review]

**Comment by squeed** [comment]
(does this change somehow prevent the CRD from being created?)

**Comment by akhilles** [comment]
> Does this change somehow prevent the CRD from being created? I forget the agent's flow for CRD creation.

I've tested this and it doesn't affect CRD creation because the operator uses `AllCiliumCRDResourceNames` in [`CreateCustomResourceDefinitions`](https://github.com/cilium/cilium/blob/9a9c9aae5ec033456612e499c8adc37766701fd7/pkg/k8s/apis/cilium.io/client/register.go#L168). I believe the split of "agent" CRDs and "all" CRDs was initially done to support this use case: https://github.com/cilium/cilium/pull/13820.

**Comment by akhilles** [comment]
Could the `dont-merge` be removed since we have some approvals and the tests are passing? :)

**Comment by akhilles** [comment]
Anything I can do to help get this merged in?

**Comment by squeed** [comment]
Just a poke was all that was needed :-)

**Pull request merged**
